### PR TITLE
feat: campaign group filter with preview

### PR DESCRIPTION
**Pull request by jeffersonrabb**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Adds preview functionality to the Campaigns Wizard Campaigns tab. At some point this functionality may make the Preview tab unnecessary, however the new approach doesn't support selecting multiple Campaign Groups so I'm leaving it untouched for now.

Dependent PRs:
~https://github.com/Automattic/newspack-plugin/pull/764~
~https://github.com/Automattic/newspack-plugin/pull/767~
~https://github.com/Automattic/newspack-plugin/pull/768~
~https://github.com/Automattic/newspack-plugin/pull/769~
~https://github.com/Automattic/newspack-plugin/pull/770~

Screen Shot 2021-01-01 at 9 37 19 AM

### How to test the changes in this Pull Request:

1. Be sure you have created some Campaign Groups and Segments.
2. Navigate to Campaign Wizard, first tab. 
3. Select a Campaign Group. Observe the "Segment to preview" `SelectControl` is shown. 
4. Select a Segment and click Preview, observe normal preview functionality.
5. Compare various preview scenarios in Preview and Campaigns tabs, verify the results are identical.

### Other information:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Comment by jeffersonrabb** [comment]
UI changes from https://github.com/Automattic/newspack-plugin/pulls merged into this branch. Per discussion with @thomasguillot preview is now a `Button` that reveals a `Popover` with the segmentation `SelectControl` in it.  Change made in https://github.com/Automattic/newspack-plugin/pull/771/commits/d45da835f8fd7b483028d5838b176624d0ac2c1c. I've done no styling - handing this over to @thomasguillot for that.

**Comment by thomasguillot** [comment]
@jeffersonrabb pushed some css changes in https://github.com/Automattic/newspack-plugin/pull/771/commits/5769beac561c65b11809520f54576fb4bd01a5ca

**Review by dkoo — COMMENTED** [review]
Looks good from functional testing. Awaiting conflict resolution to complete review.

**Comment by jeffersonrabb** [comment]
> Looks good from functional testing. Awaiting conflict resolution to complete review.

Conflicts resolved, ready for review @dkoo!

File: assets/wizards/popups/views/popup-group/index.js [hunk 1]
```diff
@@ -104,29 +108,78 @@ const PopupGroup = ( {
...
+													...segments.map( s => ( { value: s.id, label: s.name } ) ),
+												] }
+												value={ segmentId }
+												onChange={ setSegmentId }
+												label={ __( 'Segment to preview', 'newspack' ) }
+											/>
+											<Button
+												isLink
+												onClick={ () => {
+													console.log( showPreview );
```
Comment (dkoo) [inline assets/wizards/popups/views/popup-group/index.js:169]: Errant `console.log`?

**Review by dkoo — CHANGES_REQUESTED** [review]
Almost there, just found a lost `console.log` wandering about.

File: assets/wizards/popups/views/popup-group/index.js (hunk 1 above)
Comment (jeffersonrabb) [reply assets/wizards/popups/views/popup-group/index.js]: Whoops! https://github.com/Automattic/newspack-plugin/pull/771/commits/a44e7062b638683c72dd99850af1eed569c529b4

**Review by dkoo — APPROVED** [review]
Looks good now!

**Pull request merged**